src/main/java/ui/components/pickers/LabelPickerUILogic.java

```diff
@@ -8,6 +8,15 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+/**
+ * Handles the logic of the label picker UI.
+ *
+ * This class provides the following key methods:
+ * 1. processTextFieldChange: this should be called after every change in the UI's text field
+ * 2. toggleLabel
+ * 3. toggleHighlightedLabel
+ * 4. moveHighlightOnLabel
```

Comment:
Not sure if this listing of methods adds any value. Instead, add good header comments to each key method.